l1-contracts/contracts/bridge/L1SharedBridge.sol

```diff
@@ -234,7 +234,8 @@ contract L1SharedBridge is IL1SharedBridge, ReentrancyGuard, Ownable2StepUpgrade
         }
     }
 
-    function _decodeLegacyData(bytes calldata _data, address _prevMsgSender) internal returns (bytes32, bytes memory) {
+    function decodeLegacyData(bytes calldata _data, address _prevMsgSender) external returns (bytes32, bytes memory) {
+        require(msg.sender == address(this));
```

Comment:
added this so we can use try / catch and still not get the attach vector on transferFrom